### ISSUE TITLE
ssh server program - stay in program

### ISSUE THREAD

**Issue by Skeeve** [cbepx-me/Anbernic-H700-RG-xx-StockOS-Modification#35]
Hi!
Could you add the option to stay in the ssh server program?

Motivation:

I usually start the ssh server only when I need it.

Unfortunately a timeout (Screensaver?) often kicks in and I noticed that, when I enabled the SSH server and then start the program to enable/disable it, the connections do not time out.

So usually I enable and then restart it again until I transferred my files after which I disable it.

An option to "temporarily enable" would be great meaning: Enable and remain in the program until disabled.

**Comment by Skeeve** [comment]
I have a solution for this.

[ActivateSSH.zip](https://github.com/user-attachments/files/15784976/ActivateSSH.zip)

The ZIP contains:

* APPS/ActivateSSH.sh
* APPS/Imgs/ActivateSSH.png
* APPS/res/tempssh.png

When you start `ActivateSSH`, the ssh server will be enabled until you press A or F.

**Comment by philippjess** [comment]
Thats helpful, thanks!

**Comment by Skeeve** [comment]
Update for Mod 2.4

[ActivateSSH-2.4.zip](https://github.com/user-attachments/files/15859134/ActivateSSH-2.4.zip)

Issues were:

1. Update cleared `res` Directory, removing the PNG shown during activation
2. Update uses another file for key-functions

Remaining issue:

PNG does not conform to 2.4 design.

**Comment by michalpl7** [comment]
Is this one also compatible with this releases? [](https://github.com/exdial/anbernic-apps?tab=readme-ov-file#ssh-enabler)

**Comment by Skeeve** [comment]
> Is this one also compatible with this releases? https://github.com/exdial/anbernic-apps?tab=readme-ov-file#ssh-enabler

If you have cbepx mod 2.4, it should be compatible. For 2.5 I haven't checked yet.

**Comment by michalpl7** [comment]
> ping @anilkusaksiz, @philippjess, and @DANILOCHINA
> 
> Update for Mod 2.4
> 
> [ActivateSSH-2.4.zip](https://github.com/user-attachments/files/15859134/ActivateSSH-2.4.zip)
> 
> Issues were:
> 
> 1. Update cleared `res` Directory, removing the PNG shown during activation
> 2. Update uses another file for key-functions
> 
> Remaining issue:
> 
> PNG does not conform to 2.4 design.

This one is working but **A button** is not doing **exit** i had to reset console. Weird.

**Comment by Skeeve** [comment]
ARGH! I see the issue! That's a mistake I did. Will update…

**Comment by Skeeve** [comment]
Update for mod 2.5

If SSH is enabled, when started, it will not interfere with the settings. It will simply display "SSH Server enabled" and wait for you to quit. It won't disable the server!

If SSH is disabled, it will display "SSH Server active" and will disable SSH when you quit.

[Activate_SSH-2.5.zip](https://github.com/user-attachments/files/15905130/Activate_SSH-2.5.zip)

**Comment by michalpl7** [comment]
Now I could quit SSH server but this is weird I transfer or delete files on SD card using this SCP ( winscp ) and there is no effect when I plugged card to card reader all files remain and new ones didn't appear.

**Comment by Skeeve** [comment]
I don't know what's going wrong there, but it's not the fault of my script.

Please double check you used the correct path.

Where did you transfer it too with scp and where were you looking when plugged into pc?

**Comment by anilkusaksiz** [comment]
> Update for mod 2.5
> 
> If SSH is enabled, when started, it will not interfere with the settings. It will simply display "SSH Server enabled" and wait for you to quit. It won't disable the server!
> 
> If SSH is disabled, it will display "SSH Server active" and will disable SSH when you quit.
> 
> [Activate_SSH-2.5.zip](https://github.com/user-attachments/files/15905130/Activate_SSH-2.5.zip)

Works great, thanks!

**Comment by cbepx-me** [comment]
Most of the time, I need to perform ssh and handheld operations at the same time, so I don't have a dedicated ssh resident program. I usually solve ssh problems by extending or turning off the screen timeout.

**Comment by cbepx-me** [comment]
Most of the time, I need to perform ssh and handheld operations at the same time, so I don't have a dedicated ssh resident program. I usually solve ssh problems by extending or turning off the screen timeout. I will update the ssh control program in the next version. Thank you for your excellent work.

**Comment by cbepx-me** [comment]
Version 3.4 has added ssh manager functionality.

**Comment by cbepx-me** [comment]
> Great stuff, [@cbepx-me](https://github.com/cbepx-me). The only thing that confused me was the (B) option during setup phase.

Once launched, this feature can be found in the App Center - APPS.

**Comment by Skeeve** [comment]
Yes, I know. I just confused me while setting up the OS. It has no effect there. Maybe during setup you can register a press on "B" as the same as disabling? You es: It just (seems to) hang when you press (B) during setup.

**Comment by cbepx-me** [comment]
This is a benign bug and I will fix it in the next release.


---- Replied Message ----
| From | Stephan ***@***.***> |
| Date | 03/04/2025 13:45 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [cbepx-me/Anbernic-H700-RG-xx-StockOS-Modification] ssh server program - stay in program (Issue #35) |

Yes, I know. I just confused me while setting up the OS. It has no effect there. Maybe during setup you can register a press on "B" as the same as disabling? You es: It just (seems to) hang when you press (B) during setup.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Skeeve left a comment (cbepx-me/Anbernic-H700-RG-xx-StockOS-Modification#35)

Yes, I know. I just confused me while setting up the OS. It has no effect there. Maybe during setup you can register a press on "B" as the same as disabling? You es: It just (seems to) hang when you press (B) during setup.

**Comment by Skeeve** [comment]
I couldn't yet figure out how to achieve it, but I think it would be great, the temporary ssh server could also switch on wifi, if it's not really on and switch off on exit when it switched it on.
